src/realm/object-store/impl/transact_log_handler.cpp

```diff
@@ -121,7 +123,8 @@ void KVOAdapter::before(Transaction& sg)
             // We may have pre-emptively marked the column as modified if the
             // LinkList was selected but the actual changes made ended up being
             // a no-op
-            list.observer->changes.erase(list.col.value);
+            REALM_ASSERT(list.path[0].is_col_key()); // first element in path is always the col_key
+            list.observer->changes.erase(list.path[0].get_col_key().value);
```

Comment:
Something fishy here. Why is it enough to identify the list by column here. Should definition of `changes` be changed? Must be investigated further.

Reply:
it seems it goes by col_key. So I removed the concept of path from ListInfo.